### PR TITLE
[UP][TECH-444]Post Sign Up Flow

### PR DESCRIPTION
**Pull request by akhil-v-011**

### CODE REVIEW

File: src/actions/signUpAction.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { AnyAction } from 'redux'
+import { ActionSignup } from '../models/actionModels/signup'
+
+export function autoPopulateEmail(email: string): ActionSignup | AnyAction {
+  return {
+    type: 'ACTION_SIGNUP',
+    payload: email,
+  }
+}
```
Comment (JeanMariaJose) [inline src/actions/signUpAction.ts:9]: Have you tried this with backend service. When you do a create account the details should get entered in DB in Create Account service and verify if the details are present in H2 DB and see the email is auto populated. You need to remove the hardcoded portions if any

**Review by JeanMariaJose — CHANGES_REQUESTED** [review]

**Review by JeanMariaJose — APPROVED** [review]

File: src/reducers/signupReducer.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  state = initialState,
+  action: ActionSignup | AnyAction,
+) => {
+  switch (action.type) {
+    case 'ACTION_SIGNUP':
+      return { ...state, autoFillEmail: action.payload }
+    default:
+      return { ...state }
+  }
+}
```
Comment (JeanMariaJose) [inline src/reducers/signupReducer.ts:18]: pls rebase your branch with main

**Review by JeanMariaJose — CHANGES_REQUESTED** [review]

**Review by JeanMariaJose — DISMISSED** [review]

File: src/pages/sign-up/SignUp.tsx [hunk 3]
```diff
@@ -52,13 +55,18 @@ const SignUp = () => {
       console.log(data)
       CreateAccountService.getCreateAccount(data)
         .then((response) => {
+          response.data
+            ? dispatch(autoPopulateEmail(data.email))
+            : dispatch(autoPopulateEmail(''))
```
Comment (abhijithramankutty) [inline src/pages/sign-up/SignUp.tsx:60]: is it really needed to have a dispatch action with empty data ?

File: src/actions/signUpAction.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { AnyAction } from 'redux'
+import { ActionSignup } from '../models/actionModels/signup'
+
+export function autoPopulateEmail(email: string): ActionSignup | AnyAction {
```
Comment (abhijithramankutty) [inline src/actions/signUpAction.ts:4]: Action function name can be append with "Action" for better readability. i.e. can be rename this to `autoPopulateEmailAction`

**Review by abhijithramankutty — APPROVED** [review]

**Pull request merged**